File: packages/app/src/services/software/definitions/megasync.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'megasync';
	public name = 'MEGAsync';
	public category = [SoftwareCategory.FileManagement, SoftwareCategory.Utility];
	public downloadName = 'MEGAsyncSetup64.exe';
	public isArchive = false;
	public shouldCacheUrl = false;
	public requiresAdmin = false;
	public deprecated = false;
	public icon = 'megasync.png';
	public homepage = 'https://mega.io/desktop';

	public async resolveDownloadUrl() {
		return ok('https://mega.nz/MEGAsyncSetup64.exe');
	}
}

File: src/lib/software/megasync.py
from src.lib.software import BaseSoftware

class Megasync(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'MEGAsyncSetup64.exe'
        self.silent_install_args = ['/S']

    def resolve_download_url(self):
        self.url_resolved.emit('https://mega.nz/MEGAsyncSetup64.exe')
